### PR TITLE
Bugfix for Sentry Deployment

### PR DESCRIPTION
**Pull request by warbluke**
Most warfare turfs are climbable even if you're not in a trench, so you were able to escape a deployed sentry while still having it be deployed to make an invisible trap. Now you can't climb with a sentry in your hands (deployed or undeployed) and putting it on your back undeploys it. That should fix any loopholes to escape a deployed sentry without it undeploying.

### CODE REVIEW

File: code/game/gamemodes/warfare/turfs.dm [hunk 1]
```diff
@@ -97,6 +97,37 @@
...
+/turf/simulated/floor/do_climb(var/mob/living/user)
+	if(!can_climb(user))
+		return
+
+	if(istype(get_area(src), /area/warfare))//We're trying to go?
+		if(locate(/obj/item/gun/projectile/automatic/mg08) in user)//Locate the mg.
+			if(istype(usr.l_hand, /obj/item/gun/projectile/automatic/mg08) || istype(usr.r_hand, /obj/item/gun/projectile/automatic/mg08))
+				to_chat(user, "I can't climb with this in my hands!")//No you fucking don't.
+				return //Keep that mg stowed asshole.
+		//if(locate(/obj/item/storage) in user)//Gotta check storage as well.
```
Comment (mattmatterson111) [inline code/game/gamemodes/warfare/turfs.dm]: Just delete this code. Don't leave it commented out.

File: code/modules/projectiles/guns/projectile/mattguns.dm [hunk 2]
```diff
@@ -335,12 +335,106 @@
...
+		if(WEST)
+			user.pixel_x += 5
+		if(NORTH)
+			user.pixel_y -= 5
+		if(SOUTH)
+			user.pixel_y += 5
+			M.plane = ABOVE_HUMAN_PLANE
+	deployed = TRUE
+	playsound(src, 'sound/weapons/mg_deploy.ogg', 100, FALSE)
+//	can_climb(user) = FALSE
```
Comment (mattmatterson111) [inline code/modules/projectiles/guns/projectile/mattguns.dm]: Same here.

File: code/modules/projectiles/guns/projectile/mattguns.dm [hunk 3]
```diff
@@ -335,12 +335,106 @@
...
+				user.pixel_x += 5
+			if(WEST)
+				user.pixel_x -= 5
+			if(NORTH)
+				user.pixel_y += 5
+			if(SOUTH)
+				user.pixel_y -= 5
+		qdel(M) //Delete the mg structure.
+	deployed = FALSE
+	//can_climb(user) = TRUE
```
Comment (mattmatterson111) [inline code/modules/projectiles/guns/projectile/mattguns.dm]: And here.

File: code/modules/projectiles/guns/projectile/mattguns.dm [hunk 4]
```diff
@@ -335,12 +335,106 @@
...
+		. = ..()
+		if(deployed)
+			pack_up_mg(user)
+
+/obj/structure/mg08_structure //That thing that's created when you place down your mg, purely for looks.
+	name = "Deployed LMG Harbinger"
+/*
+	icon = 'icons/obj/items/mg08.dmi'
+	icon_state = "mg08_structure"
+	*/
```
Comment (mattmatterson111) [inline code/modules/projectiles/guns/projectile/mattguns.dm]: And here.

**Review by mattmatterson111 — CHANGES_REQUESTED** [review]

**Review by warbluke — COMMENTED** [review]
Fixed, sorry for inconvenience

**Comment by mattmatterson111** [comment]
We ride again.

**Pull request merged**